Lib9c/Model/Buff/BuffFactory.cs

```diff
@@ -29,14 +29,34 @@ public static Buff Get(BuffSheet.Row row)
         }
 
         public static IList<Buff> GetBuffs(
-            Skill.Skill skill, 
-            SkillBuffSheet skillBuffSheet, 
+            Skill.Skill skill,
+            SkillBuffSheet skillBuffSheet,
             BuffSheet buffSheet)
         {
             var buffs = new List<Buff>();
             if (!skillBuffSheet.TryGetValue(skill.SkillRow.Id, out var skillBuffRow))
                 return buffs;
-            
+
+            foreach (var buffId in skillBuffRow.BuffIds)
+            {
+                if (!buffSheet.TryGetValue(buffId, out var buffRow))
+                    continue;
+
+                buffs.Add(Get(buffRow));
+            }
+
+            return buffs;
+        }
+
+        public static IList<Buff> GetBuffs(
+            Skill.Arena.ArenaSkill skill,
+            SkillBuffSheet skillBuffSheet,
+            BuffSheet buffSheet)
```

Comment:
I think we can reduce code duplication by tying `ArenaSkill, Skill` an interface.

Reply:
changed it